### ISSUE TITLE
Doent's save size

### ISSUE THREAD

**Issue by alabotski** [hluk/CopyQ#1952]
MacOS
Clipboard Manager
6.1.0-gf6684d39-v6.1.0

After Close/Open CopyQ doesn't save itself size, only position

**Comment by hluk** [comment]
Do you mean that when you **restart** the application, the window size from previous run is not restored?

**Comment by alabotski** [comment]
> Do you mean that when you **restart** the application, the window size from previous run is not restored?

Yes that's right

**Comment by alabotski** [comment]
Hi any news? Really not very convenient... =(

**Comment by hluk** [comment]
Try removing `copyq_geometry.ini` file from configuration directory (to find out the config path use `copyq info config` command - it prints the main config file path which is in the same directory).

You can also try disabling geometry restoring with `copyq config restore_geometry false` to see if it works better.

If the above does not work and you use global shortcut to show the main window, there is command `copyq show LEFT TOP WIDTH HEIGHT` which overrides the position and size.

**Comment by alabotski** [comment]
1. I have deleted - copyq_geometry.ini. It did not help.
2. Can you help? Where can I enter command - `copyq config restore_geometry false`

**Comment by hluk** [comment]
> 2\. Can you help? Where can I enter command - `copyq config restore_geometry false`

Either into terminal or Action dialog (can be open from Item menu or with F5 shortcut from the main window).

**Comment by alabotski** [comment]
2. copyq config restore_geometry false - after this. Stopped recovering not only the size, but also the position. Appears in the middle of the screen, and not where it was last time

**Comment by alabotski** [comment]
The program is very good, I have been using it for a long time... But it has become too complicated for the basic functions - clipboard monitoring =(

**Comment by hluk** [comment]
> 2. copyq config restore_geometry false - after this. Stopped recovering not only the size, but also the position. Appears in the middle of the screen, and not where it was last time

Yeah, this leaves the position and size restoring on the window manager.

> 3\. opyq show LEFT TOP WIDTH HEIGHT - Why I did not understand this command ... The "LEFT" tab appeared and how to remove it? =(

Oh, sorry I posted wrong command. What I meant is to use something like a concrete position and size: `copyq showAt 1000 1500 500 600`

**Comment by alabotski** [comment]
@hluk 
hmm , nothing help me =(
After restart, it always takes such position

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/439337/173573686-9aa1f617-4bf3-4dcc-8364-bacfec491fa3.png">

**Comment by alabotski** [comment]
My actions

1. I close CopyQ
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/439337/173573881-0e95f4ab-b6a8-4f18-8721-fd28005500c6.png">

2. Then open it again... And see such picture... CopyQ remember position, but forget itself size.
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/439337/173574110-c73e09c9-2805-48c2-a941-c03e28f6f563.png">

**Comment by Chealer** [comment]
I cannot reproduce this with CopyQ 13 on Windows 11. Can you reproduce with a current version?

**Comment by alabotski** [comment]
@Chealer 
Problem was on macos

**Comment by Chealer** [comment]
@alabotski: Yes, but does your problem continue if you use CopyQ version 13?
And if so, can you provide your `copyq.ini` file?

**Comment by KAGEYAM4** [comment]
I am on linux, wayland, hyprland. Can copyq not remeber its window size and when opened at boot time, use that window size? i am floating copyq with window rule, but every time copyq fresh starts it forgets the last size that it used.

I have to use window rule to make copyq window appear with certain size, but i am curious if copyq can save and restore its size automatically.

**Comment by Chealer** [comment]
@KAGEYAM4: If you mean that the main window is maximized, that is already tracked in ticket #3412. In any case, always report at least your CopyQ version.

**Comment by hluk** [comment]
> I am on linux, wayland, hyprland. Can copyq not remeber its window size and when opened at boot time, use that window size? i am floating copyq with window rule, but every time copyq fresh starts it forgets the last size that it used.

Unrelated to the macOS issue, but CopyQ stores (and restores) window and dialog positions and sizes. Unfortunately, this may **not work on Wayland** - AFAIK, there is no protocol to handle that properly.

I have been using Wayland compositors for a while and at least the window size is restored properly (the windows opens in the middle of the screen though) - this worked on KDE and Sway.
